### PR TITLE
Add flag to disable concurrency limiting

### PR DESCRIPTION
**Pull request by shellderp**
Adds a way to tune and also turn it off, so we can turn it off until we're happy with its behaviour

### CODE REVIEW

**Review by swankjesse — COMMENTED** [review]
Do you mind inverting the boolean?

I’d prefer to see this in code which looks appropriately dangerous

    concurrency_limiter_disabled: true

**Comment by shellderp** [comment]
> Do you mind inverting the boolean?
> 
> I’d prefer to see this in code which looks appropriately dangerous
> 
> ```
> concurrency_limiter_disabled: true
> ```

I wasn't worried because of kotlin default values, but I can totally do that

**Comment by shellderp** [comment]
Also, do you have any suggestions on how to be able to override the `Limiter` used?

**Comment by swankjesse** [comment]
Yep, optional injection sucks. Maybe multibind something like this?

```
interface LimiterFactory {
  fun create(action: Action): Limiter<String>?
}
```

The first one to return non-null is used?

**Comment by shellderp** [comment]
> Yep, optional injection sucks. Maybe multibind something like this?
> 
> ```
> interface LimiterFactory {
>   fun create(action: Action): Limiter<String>?
> }
> ```
> 
> The first one to return non-null is used?

ah k, so by default none are bound and in that case we create a SimpleLimiter?

File: misk/src/test/kotlin/misk/web/interceptors/ConcurrencyLimitsInterceptorCustomLimiterTest.kt [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+import misk.web.RealNetworkChain
+import misk.web.actions.WebAction
+import okhttp3.HttpUrl.Companion.toHttpUrl
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.Test
+import java.time.Duration
+import javax.inject.Inject
+
+@MiskTest(startService = true)
+class ConcurrencyLimitsInterceptorCustomLimiterTest {
```
Comment (shellderp) [inline misk/src/test/kotlin/misk/web/interceptors/ConcurrencyLimitsInterceptorCustomLimiterTest.kt:28]: I had to basically copy the existing test to install a different module

File: misk/src/test/kotlin/misk/web/interceptors/ConcurrencyLimitsInterceptorTest.kt [hunk 2]
```diff
@@ -131,6 +141,19 @@ class ConcurrencyLimitsInterceptorTest {
...
       install(LogCollectorModule())
       install(MiskTestingServiceModule())
+
+      multibind<ConcurrencyLimiterFactory>().toInstance(CustomLimiterFactory())
+    }
+  }
+
+  class CustomLimiterFactory : ConcurrencyLimiterFactory {
+    override fun create(action: Action): Limiter<String>? {
+      if (action.function == CustomLimiterAction::call) {
```
Comment (swankjesse) [inline misk/src/test/kotlin/misk/web/interceptors/ConcurrencyLimitsInterceptorTest.kt:151]: oh cool

**Review by swankjesse — APPROVED** [review]

**Pull request merged**